apache2/apache2_config.c

```diff
@@ -1557,18 +1547,23 @@ static const char *cmd_data_dir(cmd_parms *cmd, void *_dcfg, const char *p1)
 #ifdef WAF_JSON_LOGGING_ENABLE
     strcpy( wafjsonlog_path, dcfg->data_dir );
     strcat( wafjsonlog_path, WAF_LOG_UTIL_FILE );
-    rc = apr_file_open(&dcfg->wafjsonlog_fd, wafjsonlog_path,
+    rc = apr_file_open(&msc_waf_log_fd, wafjsonlog_path,
                    APR_WRITE | APR_APPEND | APR_CREATE | APR_BINARY,
                    CREATEMODE | APR_WREAD, cmd->pool);
 
     if (rc != APR_SUCCESS) {
         return apr_psprintf(cmd->pool, "ModSecurity: Failed to open wafjson log file: %s",
             wafjsonlog_path);
     }
+    
+    // Global variable to share between threads
+    strncpy( msc_waf_log_path, wafjsonlog_path, 1024 );
```

Comment:
Let's use some common `#define` or `const` for the buffer length both here and in place where it is defined. Otherwise if, say, later someone changes it from 1024 to 512 in the future we don't expose ourselves to a buffer overrun. This will also be better because a named constant is always easier to recognize than a magic value.

Reply:
will change